src/systemctl/systemctl.c

```diff
@@ -1658,6 +1658,9 @@ static int list_unit_files(int argc, char *argv[], void *userdata) {
                 for (unit = units; unit < units + c; unit++)
                         free(unit->path);
 
+        if (c == 0)
+                return 1;
```

Comment:
Sorry I really should have mentioned this in the last comment, but please return a negative error code here per the coding style guidelines (maybe `-ENOENT`?).

Reply:
Thanks, I have done that. I thought those were reserved for functions which clobber `errno`. Will know now for the future. Also, I have rebased and squashed the commits on top of `master`. Let me know if there are any other problems.